loki/transformations/inline.py

```diff
@@ -389,6 +393,19 @@ def __init__(self, query, recurse_query=None, inline_elementals_only=False,
         def map_inline_call(self, expr, *args, **kwargs):
             if not self.visit(expr, *args, **kwargs):
                 return
+            if expr.procedure_type is BasicType.DEFERRED or isinstance(expr.routine, StatementFunction):
+                return
```

Comment:
We could add here a bailout for non-functions and save a few checks further down:
```suggestion
                return
            if not expr.procedure_type.is_function:
                return
```

Reply:
Hmm this ends up with errors like:

```
AttributeError: 'BasicType' object has no attribute 'is_function'
```

Reply:
You should add this below the checks in ll. 396-397